### DISCUSSION TITLE
Microsoft.Extensions.Logging issues

### DISCUSSION THREAD

**Discussion by jdlegan** [datalust/seq-tickets#1165] [Help]
We are a big SeriLog and Seq shop and for the first time are using the .NET Core abstractions from MS in front of SeriLog. We are seeing some odd behavior that we have yet to be able to track down. As you can see in the picture below, any attempt to add properties to the scope via the recommended dictionary model and initiating a scope with it, the properties, which consist of either ints, strings or objects all come through as simply empty arrays. The interesting part is that those that are structured objects themselves come through as a structured object of arrays, with no content but is otherwise in the correct shape.

We make heavy use of Serilogs {@} pattern for logging structure objects throughout a large code base so figuring out how to do this properly and easily, preferably without having to create scopes everywhere would be great but barring that, creating the scopes properly so the data comes through. 

Anyone have any insight into this?

**Comment by jdlegan** [comment]
![image](https://user-images.githubusercontent.com/61296514/110957235-6eb97100-8319-11eb-93df-e48163d84671.png)

**Comment by jdlegan** [comment]
Simple snippet of code taking in a non-specific JSON doc to repro the issue.
```
            var requestBody = await new StreamReader(req.Body).ReadToEndAsync();
            var o           = JObject.Parse(requestBody);
            var scopeProps  = new Dictionary<string, object> {{"OriginatingHost", req.Host}};

            foreach (var property in o.Properties())
            {
                scopeProps.Add(property.Name, property.Value);
            }

            using (log.BeginScope(scopeProps)) {

                log.LogWarning(new EventId(1), "{RequestBody}", requestBody);

                return new OkResult();
            }
```

**Reply by nblumhardt** [reply]
Hi James,

Good to hear from you. Those arrays look like the default handling of `JObject` as an `IEnumerable`; putting `property.Value.ToString()` into the scope dictionary might clean it up a little, but alternatively, plugging in _Destructurama.JsonNet_ and using `@` to prefix the property names might also help.

**Comment by jdlegan** [comment]
I had added the @ before going the JObject route but received the same results. I did not however have that nuget package in there. I will give it a try. Thank you.
